src/components/views/dialogs/BaseDialog.js

```diff
@@ -76,17 +72,22 @@ export default React.createClass({
         const TintableSvg = sdk.getComponent("elements.TintableSvg");
 
         return (
-            <div onKeyDown={this._onKeyDown} className={this.props.className}>
+            <FocusTrap onKeyDown={this._onKeyDown}
+                className={this.props.className}
+                role="dialog"
+                aria-labelledby='mx_BaseDialog_title'
+                aria-describedby={this.props.contentId}
+            >
                 <AccessibleButton onClick={this._onCancelClick}
                     className="mx_Dialog_cancelButton"
                 >
                     <TintableSvg src="img/icons-close-button.svg" width="35" height="35" />
                 </AccessibleButton>
-                <div className={'mx_Dialog_title ' + this.props.titleClass}>
+                <div className={'mx_Dialog_title ' + this.props.titleClass} id='mx_BaseDialog_title'>
```

Comment:
We need an ID not a CSS class in order to pass as an aria-labelled-by on the root node. So other than to supply the same ID at line 78 and line 86 there are no other intentions here. Perhaps it can be improved somehow but I'd appreciate a hint on how to go about doing it. Perhaps it might just be because I am not very experienced.

Reply:
Oh, I completely missed that it was an ID and not a class.